Microsoft.SystemForCrossDomainIdentityManagement/Protocol/ProtocolExtensions.cs

```diff
@@ -42,6 +53,36 @@ public static HttpMethod PatchMethod
             }
         }
 
+        public static bool IsResourceType(this object jobject, string scheme)
+        {
+            if(null == jobject)
+            {
+                throw new ArgumentNullException(nameof(jobject));
+            }
+            if (string.IsNullOrWhiteSpace(scheme))
+            {
+                throw new ArgumentNullException(nameof(scheme));
+            }
+            dynamic operationDataJson = JsonConvert.DeserializeObject(jobject.ToString());
+            var schemas = operationDataJson.schemas;
```

Comment:
var is always unfortunate, leaving the reader to wonder what the type is. One could certainly write, if (null == operationDataJson.Schemas) to eliminate it. Also one can use ... 

switch (operationDataJson.schemas)
{
    case X x: <-- I think you know what X is here, whereas I, as a reader, still don't. 
         string[] schemasList = schemas.ToObject<string[]>();
        ...
        break;
    return result;

<-- This will have handled all the cases with zero anonymous types.